lib/Core/ErrorState.cpp

```diff
@@ -772,3 +805,7 @@ ref<Expr> ErrorState::getScalingConstraint() {
       UpdateList(array, 0), ConstantExpr::create(0, array->getDomain()));
   return NeExpr::create(scalingVal, ConstantExpr::create(0, Expr::Int8));
 }
+
+std::map<std::string, ref<Expr> > ErrorState::getStateErrorExpressions() {
```

Comment:
This can be
```
std::map<std::string, ref<Expr> > &ErrorState::getStateErrorExpressions() { ...
```
(note the ampersand). The difference is that, with the ampersand, the function returns exactly the object, and not a copy of it, saving us copying operation, which will incur heavy performance penalty when the copied data structure is large.